src/opt.rs

```diff
@@ -189,6 +190,29 @@ pub struct CommonOpts {
     /// Path to an auth-file to read authentication information from
     #[clap(long, env = "RATTLER_AUTH_FILE", hide = true)]
     pub auth_file: Option<PathBuf>,
+
+    /// Channel priority to use when solving
+    #[arg(long, default_value = "strict")]
+    pub channel_priority: ChannelPriorityWrapper,
+}
+
+/// Container for rattler_solver::ChannelPriority so that it can be parsed
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub struct ChannelPriorityWrapper {
+    /// The ChannelPriority value to be used when building the Configuration
+    pub value: ChannelPriority
+}
+
+impl FromStr for ChannelPriorityWrapper {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "strict" => Ok(ChannelPriorityWrapper { value: ChannelPriority::Strict }),
+            "disabled" => Ok(ChannelPriorityWrapper { value: ChannelPriority::Disabled }),
+            _ => Err(format!("Unknown channel priority: {}", s))
```

Comment:
For a user it would be nice to list the available options for a perfectly helpful error message :)

Reply:
I've updated the error message